functionList/LSL_byKimpaTammas.xml

```diff
@@ -0,0 +1,51 @@
+<?xml version="1.0" encoding="UTF-8" ?>
+<!-- ==========================================================================\
+|
+|   To learn how to make your own language parser, please check the following
+|   link:
+|       https://npp-user-manual.org/docs/function-list/
+|
+\=========================================================================== -->
+<NotepadPlus>
+	<functionList>
+		<!-- ================================================================= [ LSL ] -->
+		<!-- LSL - Linden Script Language                                              -->
+
+		<parser
+			id="lsl"
+			displayName="LSL"
+			commentExpr="(?:(/\*.*?\*/)|(//.*?$))"
+		>
+			<classRange
+				mainExpr="(?x)(?:\b(?:(state)\s+(?:[A-Za-z_]\w*)|(default))\b)(?=\s*\{)"
+				openSymbole="\{"
+				closeSymbole="\}"
+				displayMode="node"
+			>
+				<className>
+					<nameExpr expr="\b(?:(state)\s+(?:[A-Za-z_]\w*))\b" />
+					<nameExpr expr="\b(default)\b" />
+				</className>
+				<function
+					mainExpr="(?x)(?:\b(state_(?:entry|exit)|touch(?:_(?:start|end))?|(?:land_)?collision(?:_(?:start|end))?|timer|listen|(?:no_)?sensor|control|(?:not_)?at_(?:rot_)?target|money|email|experience_permissions(?:_denied)?|run_time_permissions|changed|attach|dataserver|moving_(?:start|end)|linkset_data|link_message|(?:on|object)_rez|remote_data|http_re(?:sponse|quest)|path_update|transaction_result)\b)\s*(?:\((?:\b(?:(?:float|integer|key|list|quaternion|rotation|string|vector)\s+(?:[A-Za-z_]\w*))\b(?:\s*,\s*(?:\b(?:(?:float|integer|key|list|quaternion|rotation|string|vector)\s+(?:[A-Za-z_]\w*))\b))*)?\))(?=\s*\{)"
+					openSymbole="\{"
+					closeSymbole="\}"
+				>
+					<functionName>
+						<funcNameExpr expr="\b(state_(?:entry|exit)|touch(?:_(?:start|end))?|(?:land_)?collision(?:_(?:start|end))?|timer|listen|(?:no_)?sensor|control|(?:not_)?at_(?:rot_)?target|money|email|experience_permissions(?:_denied)?|run_time_permissions|changed|attach|dataserver|moving_(?:start|end)|linkset_data|link_message|(?:on|object)_rez|remote_data|http_re(?:sponse|quest)|path_update|transaction_result)\b" />
+					</functionName>
+				</function>
+			</classRange>
+			<function
+				mainExpr="(?x)(?:\b(?:(?:float|integer|key|list|quaternion|rotation|string|vector)\s+)?([A-Za-z_]\w*)\b)\s*\((?:\b(?:(?:float|integer|key|list|quaternion|rotation|string|vector)\s+(?:[A-Za-z_]\w*))\b(?:\s*,\s*(?:\b(?:(?:float|integer|key|list|quaternion|rotation|string|vector)\s+(?:[A-Za-z_]\w*))\b))*)?\)(?=\s*\{)"
+				openSymbole="\{"
+				closeSymbole="\}"
+				displayMode="$functionName"

```

Comment:
Is it "not supported" (ie, it can be there, but Notepad++ does nothing with it), or "not allowed" (ie, it shouldn't be there at all)?  Because the [User Manual says](https://npp-user-manual.org/docs/function-list/) that the `<function>` and `<classRange>` elements both have that attribute, but it's "reserved for future use".  And, for example, your [vb-3.0.FL.xml](https://github.com/MAPJe71/Languages/blob/master/VB%5BAS%5D%20-%20Visual%20Basic%20(for%20Applications%2C%20Script)/Config/vb-3.0.FL.xml#L15)

Reply:
It can be there, but Notepad++ does nothing with it.

> And, for example, your [vb-3.0.FL.xml](https://github.com/MAPJe71/Languages/blob/master/VB%5BAS%5D%20-%20Visual%20Basic%20(for%20Applications%2C%20Script)/Config/vb-3.0.FL.xml?rgh-link-date=2024-06-12T22%3A38%3A02Z#L15)

My Languages repository is far from up-to-date. I tend to remove the attributes, etc. that are not currently used, add comment and reformat the XML text to accommodate the comments.
The `vb-3.0.FL.xml` is an almost raw copy of a Notepad++ SourceForge topic. AFAICT I only added some comment (e.g. `L_VB` and `vb_function`).